### PR TITLE
Move bridges to separate packages

### LINKED ISSUE #61677: `clickhouse-odbc-bridge` and `clickhouse-library-bridge` should be separate packages
**Use case**

It will leave the room in the main package.

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Backward Incompatible Change

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
`clickhouse-odbc-bridge` and `clickhouse-library-bridge` are separate packages. This closes #61677.

---
### Modify your CI run:
**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Include tests (required builds will be added automatically):
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Unit tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with Analyzer
- [ ]  Add your option here

#### Exclude tests:
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with MSAN
- [ ]  All with UBSAN
- [ ]  All with Coverage
- [ ]  All with Aarch64
- [ ]  Add your option here

#### Extra options:
- [ ]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ]  1
- [ ]  2
- [ ]  3
- [ ]  4

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 89cee0a3d6a71e1fbba92ce4546cdb261b196802 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/62114/89cee0a3d6a71e1fbba92ce4546cdb261b196802/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: programs/odbc-bridge/CMakeLists.txt [hunk 1]
```diff
@@ -30,12 +30,7 @@ target_link_libraries(clickhouse-odbc-bridge PRIVATE
 set_target_properties(clickhouse-odbc-bridge PROPERTIES RUNTIME_OUTPUT_DIRECTORY ..)
 target_compile_options (clickhouse-odbc-bridge PRIVATE -Wno-reserved-id-macro -Wno-keyword-macro)
 
-if (SPLIT_DEBUG_SYMBOLS)
```
Comment (antaljanosbenjamin) [inline programs/odbc-bridge/CMakeLists.txt:33]: Do I understand correctly that we won't ship debug info for the two libs?

I assume (but not sure about it) we build `RelWithDebInfo` packages, then split the debug symbols from the binaries and ship them in separate package. With this change, we will ship the debug symbols with `clickhouse-odbc-bridge` and `clickhouse-library-bridge`, right?

Comment (alexey-milovidov) [reply programs/odbc-bridge/CMakeLists.txt:33]: Yes, all true.

**Comment by Felixoid** [comment]
Just adding packages in won't automagically deploy them anywhere. First, such changes are best to be backported so that the new packages will appear in every release. In such case, the package should be added to https://github.com/ClickHouse/clickhouse-repos-manager/blob/main/clickhouse-repos-manager/packages.py#L55

If the changes are difficult to backport, there's https://github.com/ClickHouse/clickhouse-repos-manager/blob/main/clickhouse-repos-manager/packages.py#L63, so missing packages won't fail the release deployment.

Now the packages are missing in release assets and repositories https://github.com/ClickHouse/ClickHouse/issues/63262, and breaks bugfix check https://github.com/ClickHouse/ClickHouse/actions/runs/8914292856/job/24483405669#step:8:76

**Comment by antaljanosbenjamin** [comment]
> I don't see any profit in a given action. 

I guess the only benefit is reduced package size when you don't need the bridges.

**Comment by alexey-milovidov** [comment]
There are multiple reasons for this change:

1. Allow separate builds for bridges, which will consequently allow them to be built with different libc, which is needed for fully-static ClickHouse builds.
2. First step towards deprecating the library bridge - we are not going to deprecate it now, but it will be easier in the future.

These changes should not be backported because we only backport bug fixes.

**Comment by Felixoid** [comment]
> reduced package size when you don't need the bridges
True. The size of two stripped binaries were ~300M together in 24.3

```
# ls -lh /usr/bin/clickhouse*bridge
-rwxr-xr-x 1 clickhouse-bridge clickhouse-bridge 164M Apr 30 18:27 /usr/bin/clickhouse-library-bridge
-rwxr-xr-x 1 clickhouse-bridge clickhouse-bridge 165M Apr 30 18:27 /usr/bin/clickhouse-odbc-bridge
```

> There are multiple reasons for this change:

Thank you very much, it clarifies the decision. It would really improve transparency to have it in either a linked issue or here in the PR 🙏

> These changes should not be backported because we only backport bug fixes.

On the one hand, such differences between the release packages can cause tension in the deployment infrastructure. But I need to solve it [there](https://github.com/ClickHouse/clickhouse-repos-manager/).

On the other hand, I agree. Change is too significant to break compatibility inside a single release.

**Pull request merged**